snapcraft/internal/pluginhandler/_plugin_loader.py

```diff
@@ -0,0 +1,217 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2017 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import contextlib
+import importlib
+import logging
+import sys
+
+import jsonschema
+
+import snapcraft
+from snapcraft.internal.project_loader.errors import YamlValidationError
+from snapcraft.internal import (
+    errors,
+    sources,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def load_plugin(plugin_name, part_name, project_options, properties,
```

Comment:
this should probably be split into `get_plugin_for(plugin_name)` and `init_plugin(...)`

Reply:
To summarize our conversation on rocket: yes, but the init function will be in PartsConfig to do things like process the grammar. I'll do that in future PRs where the grammar is actually processed in PartsConfig.